pkg/framework/plugins/removepodsviolatingtopologyspreadconstraint/topologyspreadconstraint.go

```diff
@@ -496,3 +493,22 @@ func matchNodeInclusionPolicies(tsc *v1.TopologySpreadConstraint, tolerations []
 	}
 	return true
 }
+
+func newTopologyConstraintSet(constraint v1.TopologySpreadConstraint, pod *v1.Pod) topologyConstraintSet {
+	if pod.Labels != nil && len(constraint.MatchLabelKeys) > 0 {
+		if constraint.LabelSelector == nil {
+			constraint.LabelSelector = &metav1.LabelSelector{}
```

Comment:
Safer to make a deep copy of `constraint` before modifying it.

Reply:
That's a good idea 👍

Reply:
I'll mirror what scheduler is doing https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/podtopologyspread/common.go#L35

Reply:
That sounds good. We should probably keep it in sync in case new features get introduced.

Reply:
Added as a separate commit but feel free to squash as you see fit. Look forward to your feedback!